src/cpp/src/image_generation/models/autoencoder_kl.cpp

```diff
@@ -182,7 +182,19 @@ AutoencoderKL::AutoencoderKL(const std::string& vae_encoder_model,
     compile(device, *extract_adapters_from_properties(properties));
 }
 
-AutoencoderKL::AutoencoderKL(const AutoencoderKL&) = default;
+AutoencoderKL::AutoencoderKL(const AutoencoderKL& rhs) = default;
+
+AutoencoderKL AutoencoderKL::clone() {
+    AutoencoderKL cloned = *this;
```

Comment:
Let's add assert:
```
OPENVINO_ASSERT((m_encoder_model != nullptr) ^ (m_encoder_request!= nullptr), "AutoencoderKL must have exactly one of m_encoder_model or m_encoder_request initialized");
```
and then simplify to:
```cpp
 if (m_encoder_model) {
        cloned.m_encoder_model = m_encoder_model->clone();
 } else {
        cloned.m_encoder_request = m_encoder_request.get_compiled_model().create_infer_request();
 }
```
Same applies to other classes as well